src/api-helper/goal/DefaultGoalImplementationMapper.ts

```diff
@@ -16,51 +16,47 @@
 
 import { Goal } from "../../api/goal/Goal";
 import { SdmGoalEvent } from "../../api/goal/SdmGoalEvent";
-import { IsolatedGoalLauncher } from "../../api/goal/support/IsolatedGoalLauncher";
 import {
     GoalFulfillment,
     GoalFulfillmentCallback,
     GoalImplementation,
+    GoalImplementationMapper,
     GoalSideEffect,
-    SdmGoalImplementationMapper,
-} from "../../api/goal/support/SdmGoalImplementationMapper";
+} from "../../api/goal/support/GoalImplementationMapper";
 import { PushListenerInvocation } from "../../api/listener/PushListener";
 
 /**
- * Concrete implementation of SdmGoalImplementationMapper
+ * Concrete implementation of GoalImplementationMapper
  */
-export class SdmGoalImplementationMapperImpl implements SdmGoalImplementationMapper {
+export class DefaultGoalImplementationMapper implements GoalImplementationMapper {
 
     private readonly implementations: GoalImplementation[] = [];
     private readonly sideEffects: GoalSideEffect[] = [];
     private readonly callbacks: GoalFulfillmentCallback[] = [];
 
-    constructor(private readonly goalLauncher: IsolatedGoalLauncher) {
-    }
-
-    public async findImplementationBySdmGoal(goal: SdmGoalEvent, inv: PushListenerInvocation): Promise<GoalImplementation> {
+    public findImplementationBySdmGoal(goal: SdmGoalEvent): GoalImplementation {
         const matchedNames = this.implementations.filter(m =>
             m.implementationName === goal.fulfillment.name &&
             m.goal.context === goal.externalKey);
 
-        const matchedGoalImplementations = [];
```

Comment:
does this mean there are no more push tests on implementations? ... or that they aren't used?

Reply:
no, they are being used and evaluated when we SetOnGoalsOnPush. I added this here again last week which was a mistake.